### PR TITLE
OCT-1825: Initial passthrough FastAPI server

### PR DESCRIPTION
**Pull request by adam-gf**
## Definition of Done

1. [ ] Acceptance criteria are met.
2. [ ] PR is manually tested before the merge by developer(s).
    - [ ] Happy path is manually checked.
3. [ ] PR is manually tested by QA when their assistance is required (1).
    - [ ] Octant Areas & Test Cases are checked for impact and updated if required (2).
4. [ ] Unit tests are added unless there is a reason to omit them.
5. [ ] Automated tests are added when required.
6. [ ] The code is merged.
7. [ ] Tech documentation is added / updated, reviewed and approved (including mandatory approval by a code owner, should such exist for changed files).
    - [ ] BE: Swagger documentation is updated.
8. [ ] When required by QA:
    - [ ] Deployed to the relevant environment.
    - [ ] Passed system tests.

---

(1) Developer(s) in coordination with QA decide whether it's required. For small tickets introducing small changes QA assistance is most probably not required.

(2) [Octant Areas & Test Cases](https://docs.google.com/spreadsheets/d/1cRe6dxuKJV3a4ZskAwWEPvrFkQm6rEfyUCYwLTYw_Cc).

### CODE REVIEW

**Review by 0xartem — COMMENTED** [review]
I looked at the code. it all looks good to me without going much into the details. I never worked with Fast API.
Questions:
- did we re-write all the back-end requests via Fast API?
- how do we test the improvements during the testing? Perfomance etc.

**Review by kgarbacinski — COMMENTED** [review]
In general - LGTM, have you also checked with Mati the corectness of functionality within Flask endpoints only (whether the middlewared views save correct data to the db etc.)?

File: backend/app/__init__.py [hunk 1]
```diff
@@ -47,7 +46,7 @@ def register_extensions(app):
     cors.init_app(app)
     db.init_app(app)
     migrate.init_app(app, db)
-    socketio.init_app(app)
+    # socketio.init_app(app)
```
[the remark this answers is not in the document]
Comment (adam-gf) [reply backend/app/__init__.py:50]: I removed the ones I'm currently comfortable removing. I left the ones that I think convey the intention, even if obvious one.

File: backend/v2/allocations/dependencies.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    epochs_subgraph: GetEpochsSubgraph,
+    projects_contracts: GetProjectsContracts,
+    settings: GetChainSettings,
+) -> SignatureVerifier:
+    return SignatureVerifier(
+        session, epochs_subgraph, projects_contracts, settings.chain_id
+    )
+
+
+GetSignatureVerifier = Annotated[SignatureVerifier, Depends(get_signature_verifier)]
```
[the remark this answers is not in the document]
Comment (adam-gf) [reply backend/v2/allocations/dependencies.py:24]: Good idea overall, but some dependencies already depend on the alias types defined above. So moving all to the bottom of the file would mean using "" on type definitions or spliting into separate files. I'd think about it in the future prs

File: backend/v2/epochs/dependencies.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    return EpochsContracts(w3, EPOCHS_ABI, settings.epochs_contract_address)  # type: ignore[arg-type]
+
+
+GetEpochsContracts = Annotated[
+    EpochsContracts,
+    Depends(get_epochs_contracts),
+]
+
+
+async def assert_allocation_window_open(
```
[the remark this answers is not in the document]
Comment (adam-gf) [reply backend/v2/epochs/dependencies.py]: Changed naming here

File: backend/v2/uniqueness_quotients/dependencies.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        default=Decimal("0.0"),
+        description="The UQ score to be returned if the user is on the timeout list.",
+    )
+
+
+def get_uq_score_settings() -> UQScoreSettings:
+    return UQScoreSettings()
+
+
+def get_uq_score_getter(
```
[the remark this answers is not in the document]
Comment (adam-gf) [reply backend/v2/uniqueness_quotients/dependencies.py:42]: We should most probably split the allocation service, but I had no clear idea as how to do it best yet. Let's chat about it

**Review by kgarbacinski — APPROVED** [review]
LGTM as the first stage, let's focus on all remaining pains before getting started with another functionalities

**Comment by adam-gf** [comment]
Due to problems with rebasing et al. We simply moved changes to separate pr and merged that one: https://github.com/golemfoundation/octant/pull/525. 
Closing as it's resolved

**Pull request closed without merging**